src/main/java/com/thealgorithms/datastructures/trees/SplayTree.java

```diff
@@ -0,0 +1,308 @@
+    }
+
+    /**
+     * Zig operation.
+     *
+     * <p>
+     * The zig operation is used to perform a single rotation on a node to move it closer to
+     * the root of the tree. It is typically applied when the node is a left child of its parent
+     * and needs to be rotated to the right.
+     * </p>
+     *
+     * @param x The node to perform the zig operation on.
+     * @return The new root node after the operation.
+     */
+    private Node rotateRight(Node x) {
+        Node y = x.left;
+        x.left = y.right;
+        y.right = x;
+        return y;
+    }
+
+    /**
+     * Zag operation.
+     *
+     * <p>
+     * The zag operation is used to perform a single rotation on a node to move it closer to
+     * the root of the tree. It is typically applied when the node is a right child of its parent
+     * and needs to be rotated to the left.
+     * </p>
+     *
+     * @param x The node to perform the zag operation on.
+     * @return The new root node after the operation.
+     */
+    private Node rotateLeft(Node x) {
+        Node y = x.right;
+        x.right = y.left;
+        y.left = x;
+        return y;
+    }
+
+    /**
+     * Splay operation.
+     *
+     * <p>
+     * The splay operation is the core operation of a splay tree. It moves a specified node
+     * closer to the root of the tree by performing a series of rotations. The goal of the splay
+     * operation is to improve the access time for frequently accessed nodes by bringing them
+     * closer to the root.
+     * </p>
+     *
+     * <p>
+     * The splay operation consists of three main cases:
+     * <ul>
+     * <li>Zig-Zig case: Perform two consecutive rotations.</li>
+     * <li>Zig-Zag case: Perform two consecutive rotations in opposite directions.</li>
+     * <li>Zag-Zag case: Perform two consecutive rotations.</li>
+     * </ul>
+     * </p>
+     *
+     * <p>
+     * After performing the splay operation, the accessed node becomes the new root of the tree.
+     * </p>
+     *
+     * @param root The root of the subtree to splay.
+     * @param key  The key to splay around.
+     * @return The new root of the splayed subtree.
+     */
+    private Node splay(Node root, int key) {
+        if (root == null || root.key == key) return root;
+
+        if (root.key > key) {
+            if (root.left == null) return root;
+            // Zig-Zig case
+            if (root.left.key > key) {
+                // Recursive call to splay on grandchild
+                root.left.left = splay(root.left.left, key);
+                // Perform zig operation on parent
+                root = rotateRight(root);
+            } // Zig-Zag case
+            else if (root.left.key < key) {
+                // Recursive call to splay on grandchild
+                root.left.right = splay(root.left.right, key);
+                // Perform zag operation on parent
+                if (root.left.right != null) root.left = rotateLeft(root.left);
```

Comment:
This condition is never `false`. Please add a tests covering such situation.

Furthermore:
```suggestion
                if (root.left.right != null) {
                    root.left = rotateLeft(root.left);
                }
```

Reply:
I have tried to make the tree to be enough complex so that it can cover this case.